File: 2024STproject/SE_Back_End/reference/DbOracle/Models/Salary.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace DbOracle.Models;

public partial class Salary
{
    public decimal SalaryId { get; set; }

    public decimal? EmpId { get; set; }

    public decimal? Year { get; set; }

    public decimal? Month { get; set; }

    public decimal? Bonus { get; set; }

    public decimal? HolidayAllowance { get; set; }

    public decimal? OtherAllowance { get; set; }

    public decimal? Commission { get; set; }

    public decimal? YearEndBonus { get; set; }

    public decimal? OvertimePay { get; set; }

    public string? RewardType { get; set; }

    public decimal? RewardAmount { get; set; }

    public decimal? LateDeduction { get; set; }

    public decimal? EarlyDepartureDeduction { get; set; }

    public decimal? IncomeTax { get; set; }

    public decimal? SocialInsurance { get; set; }

    public decimal? GrossSalary { get; set; }

    public decimal? NetSalary { get; set; }

	[JsonIgnore]
	public virtual Employee? Emp { get; set; }
}

File: 2024STproject/SE_Back_End/performance-service/src/main/java/com/jike/performance/domain/po/Salary.java
package com.jike.performance.domain.po;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serializable;
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("SALARY")
public class Salary implements Serializable{
    private static final long serialVersionUID = 1L;
    /**
     * id
     */
    @TableId(value = "Salary_ID", type = IdType.AUTO)
    private int salaryId;

    /**
     * 员工ID
     */
    @TableField("EMPLOYEE_ID")
    private int employeeId;

    /**
     * 年份
     */
    @TableField("YEAR")
    private int year;

    /**
     * 月份
     */
    @TableField("MONTH")
    private int month;

    /**
     * 绩效工资
     */
    @TableField("BONUS")
    private double bonus;

    /**
     * 节假日补贴
     */
    @TableField("HOLIDAY_ALLOWANCE")
    private double holidayAllowance;

    /**
     * 其它补贴
     */
    @TableField("OTHER_ALLOWANCE")
    private double otherAllowance;

    /**
     * 提成
     */
    @TableField("COMMISSION")
    private double commission;

    /**
     * 年终奖
     */
    @TableField("YEAR_END_BONUS")
    private double yearEndBonus;

    /**
     * 加班费
     */
    @TableField("OVERTIME_PAY")
    private double overtimePay;

    /**
     * 奖励类型
     */
    @TableField("REWARD_TYPE")
    private String rewardType;

    /**
     * 奖励金额
     */
    @TableField("REWARD_AMOUNT")
    private double rewardAmount;

    /**
     * 迟到扣款
     */
    @TableField("LATE_DEDUCTION")
    private double lateDeduction;

    /**
     * 早退扣款
     */
    @TableField("EARLY_DEPARTURE_DEDUCTION")
    private double earlyDepartureDeduction;

    /**
     * 个人所得税
     */
    @TableField("INCOME_TAX")
    private double incomeTax;


    /**
     * 五险一金
     */
    @TableField("SOCIAL_INSURANCE")
    private double socialInsurance;

    /**
     * 实发工资
     */
    @TableField("GROSS_SALARY")
    private double grossSalary;

    /**
     * 应发工资
     */
    @TableField("NET_SALARY")
    private double netSalary;

}
